packages/x-charts/src/BarChart/seriesConfig/bar/seriesProcessor.ts

```diff
@@ -66,41 +66,101 @@ const seriesProcessor: SeriesProcessor<'bar'> = (params, dataset) => {
 
   const completedSeries: {
     [id: string]: DefaultizedBarSeriesType & {
+      visibleStackedData: [number, number][];
       stackedData: [number, number][];
     };
   } = {};
 
   stackingGroups.forEach((stackingGroup) => {
     const { ids, stackingOffset, stackingOrder } = stackingGroup;
-    // Get stacked values, and derive the domain
-    const stackedSeries = d3Stack<any, DatasetElementType<number | null>, SeriesId>()
-      .keys(
-        ids.map((id) => {
-          // Use dataKey if needed and available
-          const dataKey = series[id].dataKey;
-          return series[id].data === undefined && dataKey !== undefined ? dataKey : id;
-        }),
-      )
+    const keys = ids.map((id) => {
+      // Use dataKey if needed and available
+      const dataKey = series[id].dataKey;
+      return series[id].data === undefined && dataKey !== undefined ? dataKey : id;
+    });
+
+    const stackedData = d3Stack<any, DatasetElementType<number | null>, SeriesId>()
+      .keys(keys)
       .value((d, key) => d[key] ?? 0) // defaultize null value to 0
       .order(stackingOrder)
       .offset(stackingOffset)(d3Dataset);
 
+    // We sort the keys based on the original stacking order to ensure consistency
+    const idOrder = stackedData.sort((a, b) => a.index - b.index).map((s) => s.key);
```

Comment:
Not sure about why this is needed. For me D3 takes care of the ordering with `stackingOrder`

Reply:
We have to use the exact same order as `stackedData`.

IIRC the issue was that some ordering functions change the order based on value, so if we have value `0` for a hidden series it would be ordered first/last in the stack.